### PR TITLE
added recipe entity class.

### PR DESCRIPTION
**Pull request by jdselby24**
The recipe entity class

### CODE REVIEW

File: src/RecipeEntity.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace GRUB;
+
+class RecipeEntity {
+    private $title = "";
+    private $link = "";
+    private $imageURL = "";
+    private $ingredients = "";
+
```
Comment (BuzatuCos) [inline src/RecipeEntity.php:10]: doc blocks

File: src/RecipeEntity.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return $this->title;
+    }
+    public function getLink(): string{
+        return $this->link;
+    }
+    public function getImageURL(): string{
+        return $this->imageURL;
+    }
+    public function getIngredients(): array{
+        return $this->ingredients;
```
Comment (sudokufan) [inline src/RecipeEntity.php]: consolidate into one 'generateHTML' function (with doc block)

**Review by sudokufan — CHANGES_REQUESTED** [review]

File: src/RecipeEntity.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        $html .= "<h3>$this->title</h3>";
+        $html .= "<div class='recipeImage'>";
+        $html .= "<img src='$this->imageURL'/>";
+        $html .= "</div>";
+        $html .= "<p>Ingredients: $ingredients </p>";
+        $html .= "</a>";
+        $html .= "</div>";
+
+        return $html;
+
```
Comment (BuzatuCos) [inline src/RecipeEntity.php]: spaces

File: src/RecipeEntity.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+     * @return string HTML code. 
+     */
+    public function generateHTML(): string {
+        $ingredients = "";
+
+        foreach($this->ingredients as $ingredient) {
+            $ingredients .= $ingredient .= " ";
+        }
+        
+        $html = "";
```
Comment (BuzatuCos) [inline src/RecipeEntity.php]: you don t need it

File: tests/src/RecipeEntity.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    /**
+     * Testing for malformed code. Passed in arrays not strings.
+     *
+     * @return void
+     */
+    public function testMalformedRecipeEntityGenerateHTML(){
+        $this->expectException(TypeError::class);
+        $case = new GRUB\RecipeEntity ([],[],[],9);
+
```
Comment (BuzatuCos) [inline tests/src/RecipeEntity.php]: spaces

**Review by sudokufan — APPROVED** [review]

**Pull request merged**